### PR TITLE
Add component metadata YAML for Azure CosmosDB output binding.

### PR DESCRIPTION
**Pull request by tmacam**
# Description

Add component metadata YAML for Azure CosmosDB output binding.

Verified with `make check-component-metadata && make bundle-component-metadata`

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: bindings/azure/cosmosdb/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    example: '"Orders"'
+  - name: partitionKey
+    type: string
+    required: true
+    description: |
+      The name of the key to extract from the payload (document to be created) that is used as the
+      partition key. This name must match the partition key specified upon creation of the
+      Cosmos DB container.
+    example: '"OrderId"'
+  - name: azureEnvironment
```
Comment (ItalyPaleAle) [inline bindings/azure/cosmosdb/metadata.yaml:57]: I think this is only for Azure AD

Comment (tmacam) [reply bindings/azure/cosmosdb/metadata.yaml:57]: I was wondering if that is something that we should implicitly define under `builtinAuthenticationProfiles "azuread"`.

Comment (ItalyPaleAle) [reply bindings/azure/cosmosdb/metadata.yaml:57]: uhm good point. would you mind including this change too?

Also I think the value is now case-insensitive so we can put a "LESS AGGRESSIVE" (i.e. not all caps):

- `AzurePublic`
- `AzureChina`
- `AzureUSGovernment`

Comment (tmacam) [reply bindings/azure/cosmosdb/metadata.yaml:57]: But anyway, specifically for CosmosDB, [this field is only read if authentication falls back to Azure AD(https://github.com/dapr/components-contrib/blob/3a616c5678442f3bf3e8ebeaf820f4c0a696c8e7/bindings/azure/cosmosdb/cosmosdb.go#L84-L88).

Comment (ItalyPaleAle) [reply bindings/azure/cosmosdb/metadata.yaml:57]: Yeah I do think this is only for AAD.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**